### PR TITLE
Ability to show captions and descriptions in all entered languages

### LINKED ISSUE #3949: Ability to show captions and descriptions in all entered languages
Media details currently shows only one pair of caption/description (either in the user's locale or a fallback language).

It could be useful if there was the ability to see captions and descriptions in all languages were they are available.

It would show information similar to this (probably in a more mobile-friendly form):
Screen Shot 2020-10-06 at 08 46 45

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
**Description (required)**

Fixes #3949

What changes did you make and why?

* Add ListView to show all captions of image
* Add WebView to show all description of image 
* handle Back Event of Fragment (mediaDetailFragment)
* Add Javadoc

**Tests performed (required)**
Android Device:Redmi y3(API :29),Redmi 4(API :25)

**Screenshots (for UI changes only)**
Image Link :https://commons.m.wikimedia.org/wiki/File%3AL%C3%BCdinghausen%2C_Burg_Vischering_--_2021_--_5233.jpg
Below is caption and Description
Screenshot_2021-04-13-22-58-53-568_fr free nrw commons

### CODE REVIEW

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul review my this pull request

File: app/src/main/res/layout/fragment_media_detail.xml [hunk 1]
```diff
@@ -190,6 +204,14 @@
                       tools:text="Description of the media goes here. This can potentially be fairly long, and will need to wrap across multiple lines. We hope it looks nice though." />
                 </LinearLayout>
 
+                <TextView
+                  android:id="@+id/show_caption_description_textview"
+                  style="@style/MediaDetailTextLabelGeneric"
+                  android:layout_width="match_parent"
+                  android:layout_height="wrap_content"
+                  android:layout_marginBottom="@dimen/dimen_10"
+                  android:text="Show Caption And Description in more Language" />
```
Comment (nicolas-raoul) [inline app/src/main/res/layout/fragment_media_detail.xml:213]: Please make it localizable. Also, change the text to `In all languages`, which is shorter and understandable enough. Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
When there are no captions, the `:` character looks strange:
adb
Could you please fix it? For instance just show `No captions`. Thanks!

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul it is important to add localizable for `In all languages` in all language becouse it take  much time .
First convert the string in all language then add it to string value of every language
There are 118 language string language value in the app 

Tell an better and easy way to implement localizable for `In all languages`

**Comment by nicolas-raoul** [comment]
Please perform internationalization the normal way, for instance `android:text="@string/media_detail_author"`.
You do not need to translate.

**Review by neslihanturan — CHANGES_REQUESTED** [review]
128 files are edited in this PR, it is because of edited string files for all languages. We do not edit string files contains specific languages manually. We only edit the generic strings.xml and others will be edited with translation updates. Those changes should be reverted.

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul i made the required changes review it

**Review by neslihanturan — APPROVED** [review]

**Comment by neslihanturan** [comment]
Hey, these changes are good for me. I would prefer more detailed javadocs but I am okay with this too. We can merge as soon as @nicolas-raoul also approves.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Works great with https://commons.wikimedia.org/wiki/File:Common_ceres_forester_(Euphaedra_phaethusa_phaethusa)_male.jpg in `Explore`:
adb

However, it crashes as soon as I tap "In all languages" for any image in my `Contributions`:
```
05-27 13:57:13.875 15348 15348 E AndroidRuntime: java.lang.IllegalStateException: The application's PagerAdapter changed the adapter's contents without calling PagerAdapter#notifyDataSetChanged! Expected adapter item count: 80, found: 100 Pager id: fr.free.nrw.commons:id/mediaDetailsPager Pager class: class androidx.viewpager.widget.ViewPager Problematic adapter: class fr.free.nrw.commons.media.MediaDetailPagerFragment$MediaDetailAdapter
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1143)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at androidx.viewpager.widget.ViewPager.onMeasure(ViewPager.java:1622)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7134)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:992)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.View.measure(View.java:24967)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3266)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1993)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2295)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1853)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8476)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:761)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:696)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
05-27 13:57:13.875 15348 15348 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```
Could you please test using the test account I sent you (refer to your email "Need Commons account for developers"). Please wipe the app's data in Android settings, then log in with this user and immediately tap "In all languages" for any image in `Contributions`.

Also, it seems that I always need to tap "In all languages" twice. The first tap has no visible effect.

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul i not found the cause of crash but i notice this crash occur when open the image for first time and mostly this crash not occur or sometime it occurs

**Comment by neslihanturan** [comment]
Hey @Prince-kushwaha , I think this error can be related with CaptionListViewAdapter implementation (getItem(i) returns null and getCount() returns zero) or lack of notifyDataSetChanged placement. Could you please take a look at it again considering these 2 possibilities?

**Comment by Prince-kushwaha** [comment]
@neslihanturan the solution suggest by you not work the cause of issue is `java.lang.IllegalStateException: The application's PagerAdapter changed the adapter's contents without calling PagerAdapter#notifyDataSetChanged! Expected adapter item count: 80, found: 100 Pager id: fr.free.nrw.commons:id/mediaDetailsPager Pager class: class androidx.viewpager.widget.ViewPager Problematic adapter: class fr.free.nrw.commons.media.MediaDetailPagerFragment$MediaDetailAdapter`
not the CaptionListViewAdapter implementation

**Comment by neslihanturan** [comment]
Can you please fix conflicts so that we can try it again @Prince-kushwaha ?

**Comment by Prince-kushwaha** [comment]
@neslihanturan i resolve the conflict

**Comment by Prince-kushwaha** [comment]
@nicolas-raoul i fixed all the issue of this bug you can test it

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**